_scripts/clone-l10n.sh

```diff
@@ -1,64 +1,104 @@
-if [ ! -d "fxa-content-server-l10n" ]; then
-	echo "Downloading L10N files from $DOWNLOAD_PATH..."
-	git clone --depth 1 $DOWNLOAD_PATH
+
+if [ -n "${FXA_L10N_COPY_FROM}" ]; then
+    echo "Using L10N files from: ${FXA_L10N_COPY_FROM}"
+    [ ! -d "${DOWNLOAD_PATH}" ] || rm -rf "${DOWNLOAD_PATH}"
+    ln -sf "${FXA_L10N_COPY_FROM}" "${DOWNLOAD_PATH}"
+else
+    # Git repository to clone
+    FXA_L10N_REPO="${FXA_L10N_REPO:-https://github.com/mozilla/fxa-content-server-l10n.git}"
+
+    # branch name
+    FXA_L10N_BRANCH="${FXA_L10N_BRANCH:-master}"
+
+    # number of commits to truncate the history for specific commit checkout
+    FXA_L10N_DEPTH="${FXA_L10N_DEPTH:-1000}"
+
+    clone_opts=(--depth 1)
+
+    # fetch some history when required
+    [ -z "${FXA_L10N_SHA}" ] || clone_opts=(--depth "${FXA_L10N_DEPTH}")
+
+    # directly checkout to the desired branch and don't fetch others
+    [ -z "${FXA_L10N_BRANCH}" ] || clone_opts+=(--branch "${FXA_L10N_BRANCH}" --single-branch)
+
+    # Download L10N using git
+    if [ ! -d "${DOWNLOAD_PATH}" ]; then
+        log "Downloading L10N files from: ${FXA_L10N_REPO}"
+        git clone "${clone_opts[@]}" "${FXA_L10N_REPO}" "${DOWNLOAD_PATH}"
+    fi
+
+    log "Updating L10N files in: $(realpath -m "${DOWNLOAD_PATH}")"
+
+    cd "${DOWNLOAD_PATH}"
+
+    # handle branch updates
+    if [ -n "${FXA_L10N_BRANCH}" ]; then
+        git checkout --quiet -- .
+        git checkout --quiet --force "${FXA_L10N_BRANCH}"
+        git pull --quiet origin "${FXA_L10N_BRANCH}"
+    fi
+
+    # checkout to a specific commit
+    if [ -n "${FXA_L10N_SHA}" ]; then
+        # ensure we have enough history
+        git fetch --quiet --depth "${FXA_L10N_DEPTH}"
+        git checkout --quiet --force "${FXA_L10N_SHA}"
+    fi
+    git rev-parse HEAD > git-head.txt
 fi
-cd fxa-content-server-l10n
-echo "Updating L10N files"
-git checkout -- .
-git checkout $FXA_L10N_SHA
-git pull
-git rev-parse $FXA_L10N_SHA > git-head.txt
+
+copy_ftl() {
+    local ftl_name locale_dir
+    ftl_name="$1"
+
+    for src in locale/**"/${ftl_name}.ftl"; do
+        [ -f "$src" ] || continue
+        locale_dir="$MODULE_PATH/public/locales/$(echo "$src" | sed -E 's/^.*\/([a-zA-Z_]+)\/.+/\1/; s/_/-/g')"
+        [ -d "$locale_dir" ] || mkdir -p "$locale_dir"
+        cp "$src" "$locale_dir/"
+    done
+}
 
 PAYMENTS="fxa-payments-server"
 SETTINGS="fxa-settings"
 
 # Copy .ftl files for payments or settings
-if [ "$MODULE" = "$PAYMENTS" ] || [ "$MODULE" = "$SETTINGS" ]; then
-  case "$MODULE" in
-    "$PAYMENT") FTL_FILENAME="main" ;;
-    "$SETTINGS") FTL_FILENAME="settings" ;;
-    *) FTL_FILENAME="*" ;;
-  esac
-
-  for src in locale/**/$FTL_FILENAME.ftl; do
-    [ -f "$src" ] || continue
-    dir=$(dirname "$src")
-    dir=$(echo "$dir" | sed "s/_/-/")
-    dir=$(basename "$dir")
-    base=$(basename "$src")
-    mkdir -p "../public/locales/$dir"
-    cp "$src" "../public/locales/$dir/$base"
-  done
-fi
-
-cd "$INIT_CWD"
```

Comment:
It feels sloppy to have this at the end of the script (especially with no check for an empty variable(!)) but I suspect it's there for a reason.  It looks like the `husky` package uses INIT_CWD just from grepping, but I don't know if that is related.

Reply:
This is not how shells work.

Whenever you invoke a shell script it is executed in its own process, which means that you are not modifying the invoking shell, including the working directory.

A test:
```bash
kane@kane:~$ cat > cd-test.sh <<EOF
#!/usr/bin/env bash

[ -d test ] || mkdir test
cd test
echo "pwd = $(pwd)"
EOF
kane@kane:~$ chmod +x cd-test.sh 
kane@kane:~$ pwd
/home/kane
kane@kane:~$ ./cd-test.sh
pwd = /home/kane/test
kane@kane:~$ pwd
/home/kane
```
So no, that final `cd` is absolutely useless.

Reply:
There are many different ways to run bash scripts and this is part of a larger pipeline.  I don't know how this is called, so I'm pointing out that it might have been used in a larger context that would break if it's removed.  It's something for us to check for or watch for when the patch lands.

Reply:
I think it was just sloppiness on my part.

Reply:
@clouserw This behavior is guaranteed by [POSIX](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#tag_18_12):
> A subshell environment shall be created as a duplicate of the shell environment, except that signal traps that are not being ignored shall be set to the default action. Changes made to the subshell environment shall not affect the shell environment.

The only exceptions here are [`exec`](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#exec) and [`source`/`.`](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/V3_chap02.html#dot), where `exec` will _replace_ the current execution environment and `source` executes all commands in a script _inside_ the current execution environment (and thus, will change the invokers environment).

I hope that clarifies things.